cmd/kubeadm/app/master/apiclient.go

```diff
@@ -56,17 +58,35 @@ func CreateClientAndWaitForAPI(file string) (*clientset.Clientset, error) {
 			return false, nil
 		}
 		n := &nodeList.Items[0]
-		if !v1.IsNodeReady(n) {
-			fmt.Println("[apiclient] First node has registered, but is not ready yet")
-			return false, nil
+		if v1.IsNodeReady(n) {
+			firstNodeIsReady = true
+		} else {
+			networkNotReady := false
+			for _, c := range n.Status.Conditions {
+				if c.Type == v1.NodeReady {
+					if strings.Contains(c.Message, "reason:NetworkPluginNotReady") {
```

Comment:
Any ideas on what could be a more stable way of checking?  I felt I could get away with this because the fallback if the string doesn't match is to continue waiting, which matches the current behavior.

This is kind of a band-aid to begin with though.  I think we need to revisit how/when/whether kubeadm does cluster validation, and this PR mostly has preservation of existing behavior in mind, rather than having an opinion on whether this is the best place to do it (and I think I agree with you that it's not.)